packages/flutter/lib/src/material/search.dart

```diff
@@ -476,12 +493,10 @@ class _SearchPageState<T> extends State<_SearchPage<T>> {
   Widget build(BuildContext context) {
     assert(debugCheckHasMaterialLocalizations(context));
     final ThemeData theme = widget.delegate.appBarTheme(context);
-    final String searchFieldLabel = widget.delegate.searchFieldLabel
-      ?? MaterialLocalizations.of(context).searchFieldLabel;
-    final TextStyle searchFieldStyle = widget.delegate.searchFieldStyle
-      ?? theme.inputDecorationTheme.hintStyle;
+    final String searchFieldLabel = widget.delegate.searchFieldLabel ??
```

Comment:
It is unfortunate that the delegate function was called `appBarTheme`. It really is a full `ThemeData` and with the changes below it will be effecting everything in the search UI, not just the app bar. Not sure of a better solution here though short of renaming it, so this will work for now.

Reply:
That's basically the whole issue on this, since the `ThemeData` is not being properly applied by default, there's all sorts of inconsistencies on the `SearchDelegate`. But the way this is working, `appBarTheme` should always come from a `Theme.of(context)` either way, so unless someone REALLY applies another entirely new theme there, we shouldn't have issues on this approach.